From 468dc1cc733eeccd8dc09ee85a59ec17546699f7 Mon Sep 17 00:00:00 2001
From: Florian Weimer <fweimer@redhat.com>
Date: Fri, 19 Jun 2026 18:22:20 +0200
Subject: [PATCH] resolv: Declare __p_class_syms, __p_type_syms for internal
 use

Reviewed-by: Carlos O'Donell <carlos@redhat.com>
Reviewed-by: Adhemerval Zanella <adhemerval.zanella@linaro.org>
(cherry picked from commit 360f352c9a6da545d798ef3015e73ca114f0d230)
(cherry picked from commit 349297a4ebaedf80701a7505351a23a8bdbd9808)
---
 include/resolv.h   | 5 +++++
 resolv/res_debug.c | 4 ----
 2 files changed, 5 insertions(+), 4 deletions(-)

diff --git a/include/resolv.h b/include/resolv.h
index 4dbbac3800..d5ad9994b9 100644
--- a/include/resolv.h
+++ b/include/resolv.h
@@ -70,6 +70,11 @@ libc_hidden_proto (__libc_res_nameinquery)
 extern __typeof (__res_queriesmatch) __libc_res_queriesmatch;
 libc_hidden_proto (__libc_res_queriesmatch)
 
+extern const struct res_sym __p_class_syms[];
+libresolv_hidden_proto (__p_class_syms)
+extern const struct res_sym __p_type_syms[];
+libresolv_hidden_proto (__p_type_syms)
+
 /* Variant of res_hnok which operates on binary (but uncompressed) names.  */
 bool __res_binary_hnok (const unsigned char *dn) attribute_hidden;
 
diff --git a/resolv/res_debug.c b/resolv/res_debug.c
index dab5283bbf..6f03e41432 100644
--- a/resolv/res_debug.c
+++ b/resolv/res_debug.c
@@ -389,8 +389,6 @@ p_fqname(const u_char *cp, const u_char *msg, FILE *file) {
  * that C_ANY is a qclass but not a class.  (You can ask for records of class
  * C_ANY, but you can't have any records of that class in the database.)
  */
-extern const struct res_sym __p_class_syms[];
-libresolv_hidden_proto (__p_class_syms)
 const struct res_sym __p_class_syms[] = {
   {C_IN,    (char *) "IN"},
   {C_CHAOS, (char *) "CHAOS"},
@@ -425,8 +423,6 @@ const struct res_sym __p_update_section_syms[] attribute_hidden = {
  * Names of RR types and qtypes.  The list is incomplete because its
  * size is part of the ABI.
  */
-extern const struct res_sym __p_type_syms[];
-libresolv_hidden_proto (__p_type_syms)
 const struct res_sym __p_type_syms[] = {
   {ns_t_a,      (char *) "A",     (char *) "address"},
   {ns_t_ns,     (char *) "NS",    (char *) "name server"},
-- 
2.43.7

